test/sql/window/test_window_range.test

```diff
@@ -0,0 +1,940 @@
+  );
+----
+NULL
+NULL
+NULL
+9
+9
+9
+
+query I
+SELECT sum (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS FIRST RANGE BETWEEN 1 PRECEDING AND 2 PRECEDING
+  );
+----
+9
+9
+9
+NULL
+NULL
+NULL
+
+query I
+SELECT sum (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS FIRST RANGE BETWEEN 2 FOLLOWING AND 1 FOLLOWING
+  );
+----
+9
+9
+9
+NULL
+NULL
+NULL
+
+query I
+SELECT sum (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS LAST RANGE BETWEEN 1000 PRECEDING AND 2 PRECEDING
+  );
+----
+NULL
+NULL
+1
+9
+9
+9
+
+query I
+SELECT sum (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS LAST RANGE BETWEEN 2000 FOLLOWING AND 1000 FOLLOWING
+  );
+----
+NULL
+NULL
+NULL
+9
+9
+9
+
+query I
+SELECT sum (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS FIRST RANGE BETWEEN 1000 PRECEDING AND 2000 PRECEDING
+  );
+----
+9
+9
+9
+NULL
+NULL
+NULL
+
+query I
+SELECT sum (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS FIRST RANGE BETWEEN 2000 FOLLOWING AND 1000 FOLLOWING
+  );
+----
+9
+9
+9
+NULL
+NULL
+NULL
+
+query I
+SELECT min (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS LAST RANGE BETWEEN 6 FOLLOWING AND UNBOUNDED FOLLOWING
+  );
+----
+2
+2
+2
+2
+2
+2
+
+query I
+SELECT min (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS LAST RANGE BETWEEN 1 PRECEDING AND 2 PRECEDING
+  );
+----
+NULL
+NULL
+NULL
+2
+2
+2
+
+query I
+SELECT min (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS LAST RANGE BETWEEN 2 FOLLOWING AND 1 FOLLOWING
+  );
+----
+NULL
+NULL
+NULL
+2
+2
+2
+
+query I
+SELECT min (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS FIRST RANGE BETWEEN 1 PRECEDING AND 2 PRECEDING
+  );
+----
+2
+2
+2
+NULL
+NULL
+NULL
+
+query I
+SELECT min (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS FIRST RANGE BETWEEN 2 FOLLOWING AND 1 FOLLOWING
+  );
+----
+2
+2
+2
+NULL
+NULL
+NULL
+
+query I
+SELECT min (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS LAST RANGE BETWEEN 1000 PRECEDING AND 2 PRECEDING
+  );
+----
+NULL
+NULL
+1
+2
+2
+2
+
+query I
+SELECT min (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS LAST RANGE BETWEEN 2000 FOLLOWING AND 1000 FOLLOWING
+  );
+----
+NULL
+NULL
+NULL
+2
+2
+2
+
+query I
+SELECT min (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS FIRST RANGE BETWEEN 1000 PRECEDING AND 2000 PRECEDING
+  );
+----
+2
+2
+2
+NULL
+NULL
+NULL
+
+query I
+SELECT min (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS FIRST RANGE BETWEEN 2000 FOLLOWING AND 1000 FOLLOWING
+  );
+----
+2
+2
+2
+NULL
+NULL
+NULL
+
+query I
+SELECT sum (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS LAST RANGE BETWEEN 6 FOLLOWING AND UNBOUNDED FOLLOWING
+  );
+----
+9
+9
+9
+9
+9
+9
+
+query I
+SELECT sum (a) OVER win FROM t2
+  WINDOW win AS (
+      ORDER BY b NULLS LAST RANGE BETWEEN 1 PRECEDING AND 2 PRECEDING
```

Comment:
I imagine the RANGE expression is also very useful with dates/timestamps, could you add a test for those types?